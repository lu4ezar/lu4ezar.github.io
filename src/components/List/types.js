// @flow
import type { DropResult } from "react-beautiful-dnd";
import type { Mode } from "../../types/mode";
import type { Filter } from "../../types/filter";
import type { ThunkAction } from "../../types";

export type Props = {|
  filter: Filter,
  handleClick: (id: string) => void,
  deleteTodo: (id: string) => void,
  toggleTodo: (id: string) => void,
  setMode: () => void,
  showMessage: (message: string) => ThunkAction,
  showTodo: (id: string, mode: Mode) => ThunkAction,
  onDragEnd: (result: DropResult) => void,
|};
