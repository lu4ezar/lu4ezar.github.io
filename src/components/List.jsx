/* eslint-disable react/jsx-props-no-spreading */
// @flow
import * as React from "react";
import { DragDropContext, Droppable } from "react-beautiful-dnd";
import styled, { type ReactComponentStyled } from "styled-components";
import Paper from "@material-ui/core/Paper";
import ListMaterial from "@material-ui/core/List";
import RootRef from "@material-ui/core/RootRef";
import { Typography } from "@material-ui/core";
import Fab from "@material-ui/core/Fab";
import AddIcon from "@material-ui/icons/Add";
import type { Todos } from "../types/todos";
import { ListColor } from "../utils/color";
import ListItem from "../elements/Todo";

type StyledListType = {
  isDraggingOver: boolean,
};

const StyledList: ReactComponentStyled<StyledListType> = styled(
  ({ isDraggingOver, ...other }) => <ListMaterial {...other} />
)`
  height: 90vh;
  max-height: 90vh;
  overflow: auto;
  transition: background 0.2s ease;
  background: ${({ isDraggingOver }) =>
    isDraggingOver ? ListColor.dragBackground : ListColor.background};
  display: flex;
  flex-direction: column;
  align-items: center;
  margin: 1em auto;
`;

type Props = {
  todos: Todos,
  handleClick: (id: number) => void,
  deleteTodo: (id: number) => void,
  toggleTodo: (id: number) => void,
  setMode: () => void,
  showMessage: (message: string) => void,
  showTodo: (id: number, mode: string) => void,
  onDragEnd: (result: Object) => void,
};

const List = ({
  todos,
  toggleTodo,
  deleteTodo,
  showTodo,
  onDragEnd,
  setMode,
  showMessage,
}: Props) => {
  const content = !todos.length ? (
    <Typography variant="h4" gutterBottom>
      nothing to show
    </Typography>
  ) : (
    todos.map((todo, index) => (
      <ListItem
        key={todo.id}
        index={index}
        todo={todo}
        toggleTodo={toggleTodo}
        deleteTodo={deleteTodo}
        showTodo={showTodo}
      />
    ))
  );
  return (
    <DragDropContext onDragEnd={onDragEnd}>
      <Droppable droppableId="droppable">
        {(provided, snapshot) => (
          <RootRef rootRef={provided.innerRef}>
            <Paper>
              <StyledList
                isDraggingOver={snapshot.isDraggingOver}
                {...provided.droppableProps}
              >
                {content}
                {provided.placeholder}
                <Fab
                  color="secondary"
                  onClick={setMode}
                  style={{
                    position: "absolute",
                    bottom: "1em",
                    right: "1em",
                  }}
                >
                  <AddIcon />
                </Fab>
              </StyledList>
            </Paper>
          </RootRef>
        )}
      </Droppable>
    </DragDropContext>
  );
};

export default List;
