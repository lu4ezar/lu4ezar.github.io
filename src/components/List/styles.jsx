/* eslint-disable react/jsx-props-no-spreading */
import React from "react";
import styled from "styled-components";
import { Paper, List as ListMui, Fab } from "@material-ui/core";
import { ListColor } from "../../utils/color";

export const StyledPaper = styled(Paper)`
  height: auto;
`;

export const StyledList = styled(({ isDraggingOver, ...other }) => (
  <ListMui {...other} />
))`
  height: 100%;
  flex-grow: 1;
  overflow: auto;
  transition: background 0.2s ease;
  background: ${({ isDraggingOver }) =>
    isDraggingOver && ListColor.dragBackground};
  display: flex;
  flex-direction: column;
  align-items: center;
  margin: 1em auto;
`;

export const StyledFab = styled(Fab)`
  position: fixed;
  bottom: 1em;
  right: 1em;
`;
