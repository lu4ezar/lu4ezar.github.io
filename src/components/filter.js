// @flow
import * as React from 'react';
import type { Item } from '../Item';
import FilterView from '../views/filterView';

type Props = {
	list: Array<Item>,
	prevProps: Props,
	item: Item,
	onChange: (arr: ?Array<?number>) => void,
	values: Array<string>
};

type State = {
	filterIsActive: boolean,
	priorityFilterStatus: boolean,
	priorityFilterValue: Array<string>,
	completedFilterStatus: boolean,
	completed: boolean,
	expiredFilterStatus: boolean,
	expired: boolean
};

class Filter extends React.Component<Props, State> {
	state = {
		filterIsActive: false,
		priorityFilterStatus: false,
		priorityFilterValue: ['normal'],
		completedFilterStatus: false,
		completed: true,
		expiredFilterStatus: false,
		expired: true
	};

	componentDidUpdate(prevProps: Props) {
		if (this.props.list !== prevProps.list && this.state.filterIsActive) {
			this.updateFilteredList();
		}
	}

	countItems = (
		property: string,
		value: string,
		list: Array<Item> = this.props.list
	): number => {
		list = list.filter(
			(item: Class<Item>): boolean => {
				return item[property] === (value: string);
			}
		);
		return list.length;
	};

	filterList = (list: Array<Item>): Array<Item> => {
		if (
			this.state.priorityFilterStatus ||
			this.state.completedFilterStatus ||
			this.state.expiredFilterStatus
		) {
			list = list.filter(this.filterItem);
		}
		return list;
	};

	filterItem = (item: Class<Item>): boolean => {
		let { priority, status } = item;
		let result = false;
		if (this.state.priorityFilterStatus) {
			if (!this.state.priorityFilterValue.includes(priority)) {
				return false;
			}
			result = true;
		}
		if (this.state.completedFilterStatus) {
			if (status === 'completed') {
				return this.state.completed;
			}
			result = !this.state.completed;
		}
		if (this.state.expiredFilterStatus) {
			if (status === 'expired') {
				return this.state.expired;
			}
			result = !this.state.expired;
		}
		return result;
	};

	updateFilteredList = (): void => {
		const list: Array<Item> = [...this.props.list];
		const filteredList: ?Array<Item> = this.state.filterIsActive
			? this.filterList(list)
			: null;
		const filteredListIdArray: ?Array<?number> = filteredList
			? filteredList.map(
					({ id }): number | null => {
						if (id != null) {
							return id;
						}
						return null;
					}
			  )
			: null;
		this.props.onChange(filteredListIdArray);
	};

	handleSelectChange = (values: { value: string, label: string }[]): void => {
		const arr = values.map(({ value }) => value);
		this.setState(
			{
				priorityFilterValue: arr
			},
			() => this.updateFilteredList()
		);
	};

	onSwitch = (checked: boolean, event: ?Event, id: string): void => {
		this.setState(
			{
				[id]: checked
			},
			() => this.updateFilteredList()
		);
	};

	render() {
		let list = [...this.props.list];
		const completedCount = this.countItems('status', 'completed', list);
		const expiredCount = this.countItems('status', 'expired', list);
		return (
			<FilterView
				{...this.state}
				onSwitch={this.onSwitch}
				handleChange={this.handleSelectChange}
				completedCount={completedCount}
				expiredCount={expiredCount}
			/>
		);
	}
}

export default Filter;
