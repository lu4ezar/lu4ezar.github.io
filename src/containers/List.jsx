// @flow
import React from "react";
import { useQuery } from "@apollo/client";
import WrappedList from "../components/List";
import { GET_TODOS } from "../apollo/queries";
import filterTodo from "../utils/filterTodo";
import { filterVar } from "../apollo/cache";

export default () => {
  const { data: { todos = [] } = {}, loading, error } = useQuery(GET_TODOS);

  const filter = filterVar();

  const visibleList = todos.filter((todo) =>
    filter.master.completed ? filterTodo(todo, filter) : todo
  );
  return <WrappedList list={visibleList} loading={loading} error={error} />;
};
