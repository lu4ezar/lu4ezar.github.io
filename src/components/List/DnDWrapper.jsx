/* eslint-disable react/jsx-props-no-spreading */
// @flow
import * as React from "react";
import { Paper } from "@material-ui/core";
import { Droppable } from "react-beautiful-dnd";
import List from "./List";
import type { Props } from "./types";

const DnDWrapper = (props: Props): React.Node => (
  <Droppable droppableId="droppable">
    {(provided, snapshot) => (
      <Paper ref={provided.innerRef}>
        <List {...props} provided={provided} snapshot={snapshot} />
      </Paper>
    )}
  </Droppable>
);

export default DnDWrapper;
