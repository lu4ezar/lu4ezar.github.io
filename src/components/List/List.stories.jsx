import React from "react";
import List from "./List";
import * as TodoStories from "../Todo/Todo.stories";

export default {
  component: List,
  title: "List",
  decorators: [(story) => <div style={{ padding: "3rem" }}>{story()}</div>],
};

const Template = (args) => <List {...args} />;

export const Default = Template.bind({});
Default.parameters = {
  apolloClient: {
    mocks: [
      {
        request: {
          query: GET_ALL,
        },
        result: {
          data: {
            todos: [lowPriorTodo, normalPriorTodo],
            checklists: [normalPriorChecklist],
          },
        },
      },
    ],
  },
};

/* export const WithPinnedTasks = Template.bind({});
WithPinnedTasks.args = {
  // Shaping the stories through args composition.
  // Inherited data coming from the Default story.
  tasks: [
    ...Default.args.tasks.slice(0, 5),
    { id: '6', title: 'Task 6 (pinned)', state: 'TASK_PINNED' },
  ],
}; */

export const Loading = Template.bind({});
Loading.args = {
  todos: [],
  loading: true,
};

export const Empty = Template.bind({});
Empty.args = {
  // Shaping the stories through args composition.
  // Inherited data coming from the Loading story.
  ...Loading.args,
  loading: false,
};
