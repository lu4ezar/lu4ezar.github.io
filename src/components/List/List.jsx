/* eslint-disable react/jsx-props-no-spreading */
// @flow
import * as React from "react";
import { DragDropContext, Droppable } from "react-beautiful-dnd";
import { Typography, LinearProgress } from "@material-ui/core";
import { Alert } from "@material-ui/lab";
import { useQuery } from "@apollo/client";
import type { Todos } from "../../types/todos";
import ListItem from "../Todo";
import { GET_TODOS } from "../../apollo/queries";
import type { Props } from "./types";
import { StyledPaper, StyledList } from "./styles";
import { filterList } from "../../redux/selectors";

const List = ({
  filter,
  toggleTodo,
  deleteTodo,
  showTodo,
  onDragEnd,
}: Props): React.Node => {
  const { data, loading, error } = useQuery(GET_TODOS);
  if (loading) return <LinearProgress />;
  if (error)
    return (
      <Alert variant="filled" severity="error">
        Error
      </Alert>
    );
  let { todos = [] }: { todos: Todos } = data ?? {};

  if (filter.master.status) {
    todos = filterList(todos, filter);
  }

  const content = todos.length ? (
    todos.map((todo, index) => (
      <ListItem
        key={todo.id}
        index={index}
        todo={todo}
        toggleTodo={toggleTodo}
        deleteTodo={deleteTodo}
        showTodo={showTodo}
      />
    ))
  ) : (
    <Typography variant="h4" gutterBottom>
      nothing to show
    </Typography>
  );

  return (
    <DragDropContext onDragEnd={onDragEnd}>
      <Droppable droppableId="droppable">
        {(provided, snapshot) => (
          <StyledPaper ref={provided.innerRef}>
            <StyledList
              isDraggingOver={snapshot.isDraggingOver}
              {...provided.droppableProps}
            >
              {content}
              {provided.placeholder}
            </StyledList>
          </StyledPaper>
        )}
      </Droppable>
    </DragDropContext>
  );
};

export default List;
