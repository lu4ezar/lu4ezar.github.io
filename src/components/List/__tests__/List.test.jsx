/* eslint-disable react/jsx-props-no-spreading */
// @flow
import React from "react";
import {
  render,
  screen,
  waitForElementToBeRemoved,
} from "@testing-library/react";
import { MockedProvider } from "@apollo/client/testing";
import List from "../List";
import { initialState as filter } from "../../../redux/reducers/filter";
import { TodoPriorityValues } from "../../../generated/graphql";
import {
  lowPriorTodo,
  normalPriorTodo,
  mocks as okMocks,
  errorMock,
  emptyMock,
  undefinedMock,
} from "../../../utils/test-utils";

const initialProps = {
  deleteTodo: jest.fn(),
  filter,
  handleClick: jest.fn(),
  onDragEnd: jest.fn(),
  setMode: jest.fn(),
  showMessage: jest.fn(),
  showTodo: jest.fn(),
  toggleTodo: jest.fn(),
};

const renderComponent = (mocks = okMocks, props = initialProps) =>
  render(
    <MockedProvider
      mocks={mocks}
      addTypename={false}
      defaultOptions={{
        watchQuery: { fetchPolicy: "no-cache" },
        query: { fetchPolicy: "no-cache" },
      }}
    >
      <List {...props} />
    </MockedProvider>
  );

describe("List", () => {
  it("renders in loading state initially", () => {
    const { container } = renderComponent();
    expect(container).toMatchSnapshot();
    expect(screen.getByRole("progressbar")).toBeInTheDocument();
  });

  it("renders list of todos after loading", async () => {
    const { container } = renderComponent();
    await waitForElementToBeRemoved(screen.getByRole("progressbar"));
    expect(container).toMatchSnapshot();
    expect(screen.getByText(lowPriorTodo.title)).toBeInTheDocument();
    expect(screen.getByText(normalPriorTodo.title)).toBeInTheDocument();
  });

  it("filters list if filter is enabled", async () => {
    const enabledFilterLowPriority = {
      ...filter,
      master: {
        status: true,
      },
      priority: {
        status: true,
        value: [TodoPriorityValues.Low],
      },
    };
    const enabledFilterProps = {
      ...initialProps,
      filter: enabledFilterLowPriority,
    };
    renderComponent(undefined, enabledFilterProps);
    await waitForElementToBeRemoved(screen.getByRole("progressbar"));
    expect(screen.getByText(lowPriorTodo.title)).toBeInTheDocument();
    expect(screen.queryByText(normalPriorTodo.title)).not.toBeInTheDocument();
  });

  it("shows 'nothing to show' message when todos array is empty", async () => {
    const enabledFilterHighPriority = {
      ...filter,
      master: {
        status: true,
      },
      priority: {
        status: true,
        value: [TodoPriorityValues.High],
      },
    };
    const enabledFilterProps = {
      ...initialProps,
      filter: enabledFilterHighPriority,
    };
    const { container } = renderComponent(emptyMock, enabledFilterProps);
    await waitForElementToBeRemoved(screen.getByRole("progressbar"));
    expect(container).toMatchSnapshot();
    expect(screen.getByText(/nothing to show/i)).toBeInTheDocument();
  });

  it("shows error message", async () => {
    const { container } = renderComponent(errorMock);
    await waitForElementToBeRemoved(screen.getByRole("progressbar"));
    expect(container).toMatchSnapshot();
    expect(screen.getByText(/error/i)).toBeInTheDocument();
  });

  it("should render without error if 'todos' is undefined", async () => {
    renderComponent(undefinedMock);
    await waitForElementToBeRemoved(screen.getByRole("progressbar"));
  });

  it("should render without error if 'data' is undefined", async () => {
    renderComponent(undefinedMock);
    await waitForElementToBeRemoved(screen.getByRole("progressbar"));
  });
});
