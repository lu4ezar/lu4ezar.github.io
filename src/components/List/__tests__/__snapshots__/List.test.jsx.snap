// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List renders in loading state initially 1`] = `
<div>
  <div
    class="MuiLinearProgress-root MuiLinearProgress-colorPrimary MuiLinearProgress-indeterminate"
    role="progressbar"
  >
    <div
      class="MuiLinearProgress-bar MuiLinearProgress-barColorPrimary MuiLinearProgress-bar1Indeterminate"
    />
    <div
      class="MuiLinearProgress-bar MuiLinearProgress-bar2Indeterminate MuiLinearProgress-barColorPrimary"
    />
  </div>
</div>
`;

exports[`List renders list of todos after loading 1`] = `
<div>
  <div
    class="MuiPaper-root sc-hKgILt svXJU MuiPaper-elevation1 MuiPaper-rounded"
  >
    <ul
      class="MuiList-root sc-eCssSg gSqQGt MuiList-padding"
      data-rbd-droppable-context-id="0"
      data-rbd-droppable-id="droppable"
    >
      <li
        aria-describedby="rbd-hidden-text-0-hidden-text-0"
        class="MuiListItem-root sc-dlfnbm dABzmH MuiListItem-gutters"
        data-rbd-drag-handle-context-id="0"
        data-rbd-drag-handle-draggable-id="1"
        data-rbd-draggable-context-id="0"
        data-rbd-draggable-id="1"
        draggable="false"
        role="button"
        tabindex="0"
        title="DoubleClick to view details"
      >
        <span>
          hope
        </span>
        <div
          class="MuiToolbar-root MuiToolbar-regular sc-gsTCUz TOrPC MuiToolbar-gutters"
        >
          <button
            class="MuiButtonBase-root MuiIconButton-root sc-bdfBwQ kBsyEI"
            tabindex="0"
            title="View details"
            type="button"
          >
            <span
              class="MuiIconButton-label"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 4.5C7 4.5 2.73 7.61 1 12c1.73 4.39 6 7.5 11 7.5s9.27-3.11 11-7.5c-1.73-4.39-6-7.5-11-7.5zM12 17c-2.76 0-5-2.24-5-5s2.24-5 5-5 5 2.24 5 5-2.24 5-5 5zm0-8c-1.66 0-3 1.34-3 3s1.34 3 3 3 3-1.34 3-3-1.34-3-3-3z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
          <button
            class="MuiButtonBase-root MuiIconButton-root sc-bdfBwQ kBsyEI"
            tabindex="0"
            title="Edit"
            type="button"
          >
            <span
              class="MuiIconButton-label"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34a.9959.9959 0 00-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
          <button
            class="MuiButtonBase-root MuiIconButton-root sc-bdfBwQ kBsyEI"
            tabindex="0"
            title="Mark as Completed"
            type="button"
          >
            <span
              class="MuiIconButton-label"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
          <button
            class="MuiButtonBase-root MuiIconButton-root sc-bdfBwQ kBsyEI"
            tabindex="0"
            title="Delete Todo"
            type="button"
          >
            <span
              class="MuiIconButton-label"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
      </li>
      <li
        aria-describedby="rbd-hidden-text-0-hidden-text-0"
        class="MuiListItem-root sc-dlfnbm dABzmH MuiListItem-gutters"
        data-rbd-drag-handle-context-id="0"
        data-rbd-drag-handle-draggable-id="2"
        data-rbd-draggable-context-id="0"
        data-rbd-draggable-id="2"
        draggable="false"
        role="button"
        tabindex="0"
        title="DoubleClick to view details"
      >
        <span>
          bliss
        </span>
        <div
          class="MuiToolbar-root MuiToolbar-regular sc-gsTCUz TOrPC MuiToolbar-gutters"
        >
          <button
            class="MuiButtonBase-root MuiIconButton-root sc-bdfBwQ kBsyEI"
            tabindex="0"
            title="View details"
            type="button"
          >
            <span
              class="MuiIconButton-label"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 4.5C7 4.5 2.73 7.61 1 12c1.73 4.39 6 7.5 11 7.5s9.27-3.11 11-7.5c-1.73-4.39-6-7.5-11-7.5zM12 17c-2.76 0-5-2.24-5-5s2.24-5 5-5 5 2.24 5 5-2.24 5-5 5zm0-8c-1.66 0-3 1.34-3 3s1.34 3 3 3 3-1.34 3-3-1.34-3-3-3z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
          <button
            class="MuiButtonBase-root MuiIconButton-root sc-bdfBwQ kBsyEI"
            tabindex="0"
            title="Edit"
            type="button"
          >
            <span
              class="MuiIconButton-label"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34a.9959.9959 0 00-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
          <button
            class="MuiButtonBase-root MuiIconButton-root sc-bdfBwQ kBsyEI"
            tabindex="0"
            title="Mark as Completed"
            type="button"
          >
            <span
              class="MuiIconButton-label"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
          <button
            class="MuiButtonBase-root MuiIconButton-root sc-bdfBwQ kBsyEI"
            tabindex="0"
            title="Delete Todo"
            type="button"
          >
            <span
              class="MuiIconButton-label"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
      </li>
    </ul>
  </div>
</div>
`;

exports[`List shows 'nothing to show' message when todos array is empty 1`] = `
<div>
  <div
    class="MuiPaper-root sc-hKgILt svXJU MuiPaper-elevation1 MuiPaper-rounded"
  >
    <ul
      class="MuiList-root sc-eCssSg gSqQGt MuiList-padding"
      data-rbd-droppable-context-id="2"
      data-rbd-droppable-id="droppable"
    >
      <h4
        class="MuiTypography-root MuiTypography-h4 MuiTypography-gutterBottom"
      >
        nothing to show
      </h4>
    </ul>
  </div>
</div>
`;

exports[`List shows error message 1`] = `
<div>
  <div
    class="MuiPaper-root MuiAlert-root MuiAlert-filledError MuiPaper-elevation0"
    role="alert"
  >
    <div
      class="MuiAlert-icon"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M11 15h2v2h-2zm0-8h2v6h-2zm.99-5C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
        />
      </svg>
    </div>
    <div
      class="MuiAlert-message"
    >
      Error
    </div>
  </div>
</div>
`;
